src/ray/object_manager/plasma/store.h

```diff
@@ -281,6 +281,9 @@ class PlasmaStore {
   /// Timer for printing debug information.
   mutable std::shared_ptr<boost::asio::deadline_timer> stats_timer_ GUARDED_BY(mutex_);
 
+  /// Timer for metric exporting.
+  mutable std::shared_ptr<boost::asio::deadline_timer> metrics_timer_ GUARDED_BY(mutex_);
```

Comment:
Does it have to be shared ptr? maybe it can just be `boost::asio::deadline_timer metrics_timer_`?

Reply:
maybe not - i didn't look in great details of why this was this actually.

Reply:
I agree with you that this is probably not needed - i think Chen had a previous PR thatt touches on some of the data race and the locking here. https://github.com/ray-project/ray/pull/18312

Maybe this could be something to be refactored in a seperate PR - I am leaning towards sticking with the existing code here.